### PR TITLE
sp/gp sbc flow - make completing party editable

### PR DESCRIPTION
**Pull request by kialj876**
Signed-off-by: Kial Jinnah <kialjinnah@gmail.com>

*Issue #:* /bcgov/entity#13822

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/common/CompletingParty.vue [hunk 1]
```diff
@@ -58,6 +60,18 @@ export default class CompletingParty extends Vue {
...
+  get isSbcStaff (): boolean {
+    const currentAccount = sessionStorage.getItem(SessionStorageKeys.CurrentAccount)
+    if (!currentAccount) return false
+    try {
+      return JSON.parse(currentAccount).accountType === AccountTypes.STAFF_SBC
+    } catch (error) {
+      console.log('Error parsing current account =', error)
+      return false
+    }
+  }
```
Comment (severinbeauvais) [inline src/components/common/CompletingParty.vue:73]: Most of this code is already in App.vue

I think that all you need is a new getter like this in state-getters.ts:
```
/** Whether the current account is SBC Staff. */
export const isPremiumAccount = (state: StateIF): boolean => {
  return (state.stateModel.accountInformation.accountType === AccountTypes.SBC_STAFF)
}
```

File: src/components/common/CompletingParty.vue [hunk 2]
```diff
@@ -58,6 +60,18 @@ export default class CompletingParty extends Vue {
     return (this.validate && !this.completingPartyValid)
   }
 
+  // FUTURE: pull this from shared auth composable (after vue3 upgrade)
+  get isSbcStaff (): boolean {
+    const currentAccount = sessionStorage.getItem(SessionStorageKeys.CurrentAccount)
+    if (!currentAccount) return false
+    try {
+      return JSON.parse(currentAccount).accountType === AccountTypes.STAFF_SBC
```
Comment (severinbeauvais) [inline src/components/common/CompletingParty.vue:68]: Weird that you're not getting an error here with the wrong AccountTypes enum (STAFF_SBC vs SBC_STAFF).

File: src/components/common/CompletingParty.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/components/common/CompletingParty.vue:73]: https://github.com/bcgov/business-edit-ui/blob/e1db511ab234c9b6a4abbd27f180c270d502c432/src/App.vue#L713

Comment (kialj876) [reply src/components/common/CompletingParty.vue:73]: ah nice okay will update

File: src/components/common/CompletingParty.vue (hunk 2 above)
Comment (kialj876) [reply src/components/common/CompletingParty.vue:68]: ah I switched the enum name name after testing it in the UI (I had it different in the create ui)

**Review by severinbeauvais — APPROVED** [review]
LGTM

**Pull request merged**